src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -434,3 +480,202 @@ def get_pipeline_run_metadata(
+
+        # Fetch the status of the _PipelineExecution
+        if METADATA_ORCHESTRATOR_RUN_ID in run.run_metadata:
+            run_id = run.run_metadata[METADATA_ORCHESTRATOR_RUN_ID].value
+        elif run.orchestrator_run_id is not None:
+            run_id = run.orchestrator_run_id
+        else:
+            raise ValueError(
+                "Can not find the orchestrator run ID, thus can not fetch "
+                "the status."
+            )
+        status = sagemaker_client.describe_pipeline_execution(
+            PipelineExecutionArn=run_id
+        )["PipelineExecutionStatus"]
+
+        # Map the potential outputs to ZenML ExecutionStatus. Potential values:
+        # https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/PipelineState
+        if status in ["Executing", "Stopping"]:
+            return ExecutionStatus.RUNNING
+        elif status in ["Stopped", "Failed"]:
+            return ExecutionStatus.FAILED
+        elif status in ["Succeeded"]:
+            return ExecutionStatus.COMPLETED
+        else:
+            raise ValueError("Unknown status for the pipeline execution.")
+
+    def generate_metadata(
+        self, execution: Any
+    ) -> Iterator[Dict[str, MetadataType]]:
+        """Generate run metadata based on the generated Sagemaker Execution.
+
+        Args:
+            execution: The corresponding _PipelineExecution object.
+
+        Yields:
+            A dictionary of metadata related to the pipeline run.
+        """
+        # Metadata
+        metadata: Dict[str, MetadataType] = dict()
+
+        # Orchestrator Run ID
+        if run_id := self._generate_orchestrator_run_id(execution):
+            metadata[METADATA_ORCHESTRATOR_RUN_ID] = run_id
+
+        # URL to the Sagemaker's pipeline view
+        if orchestrator_url := self._generate_orchestrator_url(execution):
+            metadata[METADATA_ORCHESTRATOR_URL] = Uri(orchestrator_url)
+
+        # URL to the corresponding CloudWatch page
+        if logs_url := self._generate_orchestrator_logs_url(execution):
+            metadata[METADATA_ORCHESTRATOR_LOGS_URL] = Uri(logs_url)
+
+        yield metadata
+
+    @staticmethod
+    def _generate_orchestrator_url(
+        pipeline_execution: Any,
+    ) -> Optional[str]:
+        """Generate the Orchestrator Dashboard URL upon pipeline execution.
+
+        Args:
+            pipeline_execution: The corresponding _PipelineExecution object.
+
+        Returns:
+             the URL to the dashboard view in SageMaker.
+        """
+        try:
+            region_name, pipeline_name, execution_id = (
+                dissect_pipeline_execution_arn(pipeline_execution.arn)
+            )
+
+            # Get the Sagemaker session
+            session = pipeline_execution.sagemaker_session
+
+            # List the Studio domains and get the Studio Domain ID
+            # TODO: Solve this with the config
+            domains_response = session.sagemaker_client.list_domains()
```

Comment:
What's the issue with using any domain? That it won't work? Or lead them to a broken link?
Somehow not a huge fan of having a config attribute just to make a link work

Reply:
It would work as long as they have at least one domain. In this case, we take the first domain and then use it to create the link. I am not sure how people are scoping domains though, I lack the Sagemaker UX knowledge here a bit. I thought a config attribute would help them to specify which domain to use but I am also OK with leaving it like this.

Reply:
I guess right now it would just fail if they have 0 domains, if that's a possibility?

Don't have a real opinion here though, do whatever you think makes sense, but if you don't know what makes sense, maybe leave it out for now and see if people complain?